### PR TITLE
add EmailJobErrorHandler

### LINKED ISSUE #1440: Add EmailJobErrorHandler to support sending email when job error.
To support for sending the emails when the job executes with errors, we decide to do some extensions for JobErrorHandler.
- create a new module - elasticjob-error-handler/elasticjob-error-handler-email
- load config from System parameters.
- load config from specified conf.

### PR DESCRIPTION
**Pull request by luky116**
Fixes #1440 .

Changes proposed in this pull request:
- add EmailJobErrorHandler
-
-

### CODE REVIEW

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    
+    private Integer port;
+
+    private String username;
+
+    private String password;
+
+    private String protocol = "smtp";
+
+    private Charset encoding = Charset.forName("UTF-8");
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java:41]: Maybe using java.nio.charset.StandardCharsets.UTF_8 is better?

**Review by TeslaCN — CHANGES_REQUESTED** [review]

**Comment by TeslaCN** [comment]
The way to configure this error handler is a little strange .I don't think users who are not familiar with the source code know how to use this error hander.

**Review by TeslaCN — COMMENTED** [review]
The way to configure this error handler is a little strange. I don't think users who are not familiar with the source code know how to use this error handler.

**Review by Technoboy- — COMMENTED** [review]
Forget to submit the SPI service

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java (hunk 1 above)
Comment (Technoboy-) [reply elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java:41]: Yes, name it UTF_8

**Comment by TeslaCN** [comment]
> The way to configure this error handler is a little strange. I don't think users who are not familiar with the source code know how to use this error handler.

Consider using system properties or environment variables?

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java (hunk 1 above)
Comment (luky116) [reply elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/pojo/EmailConfigurationPOJO.java:41]: I'll try this

**Comment by luky116** [comment]
> Forget to submit the SPI service

i will do this

**Pull request closed without merging**